### PR TITLE
aruha 846: validate replication command

### PR DESCRIPTION
**Pull request by adyach**
aruha 846

### CODE REVIEW

File: bubuku/cli.py [hunk 1]
```diff
@@ -227,5 +227,30 @@ def show_stats():
         _print_table(table)
 
 
+@cli.group(name='validate', help='Validates internal structures of kafka/zk')
+def validate():
+    pass
+
+
+@validate.command('replication', help='Checks zk structure for correct replica list')
```
Comment (adyach) [inline bubuku/cli.py]: As there were no particular requirements for that feature, I think we should keep it as simple as possible, and just check for validity, once rebalance is finished (can be observed on the metrics)

File: bubuku/cli.py [hunk 2]
```diff
@@ -227,5 +227,30 @@ def show_stats():
...
+        brokers = {int(x) for x in zookeeper.get_broker_ids()}
+        table = []
+        for topic_name, partition, state in zookeeper.load_partition_states():
+            if len(state['isr']) != factor or not set(state['isr']).issubset(brokers):
+                table.append({
+                    'Topic': topic_name,
+                    'State': state
+                })
+        if table:
+            _LOG.info('Invalid topics:')
```
Comment (adyach) [inline bubuku/cli.py]: The same as the previous comment, I think it makes sense to run it after rebalance is finished to be sure we have valid data in zookeeper.

File: bubuku/cli.py (hunk 1 above)
Comment (lmontrieux) [inline bubuku/cli.py]: @adyach maybe the help should be clearer, then. I would suggest `Returns all partitions whose ISR size differs from the replication factor`.

Comment (adyach) [inline bubuku/cli.py]: @lmontrieux b44932a

**Comment by lmontrieux** [comment]
👍

**Comment by adyach** [comment]
:

**Comment by antban** [comment]
:+1:

**Comment by adyach** [comment]
👍

**Pull request merged**